internal/ethapi/api.go

```diff
@@ -773,6 +787,37 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNr rpc.BlockNumb
 	} else {
 		addr = *args.From
 	}
+	// Override the fields of specified contracts before execution.
+	if accounts != nil {
+		for addr, account := range accounts {
+			// Override account nonce.
+			if account.Nonce != nil {
+				state.SetNonce(addr, uint64(*account.Nonce))
+			}
+			// Override account(contract) code.
+			if account.Code != nil {
+				state.SetCode(addr, *account.Code)
+			}
+			// Override account balance.
+			if account.Balance != nil {
+				state.SetBalance(addr, (*big.Int)(*account.Balance))
+			}
+			if account.State != nil && account.StateDiff != nil {
+				return nil, 0, false, errors.New("can't use state and stateDiff at the same time")
```

Comment:
Perhaps replace the error.New with `fmt.Errorf("account %s has both `state` and `stateDiff`", addr.Hex())`.